src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -174,6 +174,13 @@ export default function StreamOrchestrator(
         if (!enableOutOfBoundsCheck || !isOutOfPeriodList(time)) {
           return;
         }
+        const nextPeriod =
+          manifest.getPeriodForTime(time) ?? manifest.getNextPeriod(time);
+        if (!isNullOrUndefined(nextPeriod) && periodList.has(nextPeriod)) {
+          // Last check just for resilience reasons that the wanted Period is
+          // not one of the already-handled ones
+          return;
+        }
```

Comment:
NOTE: This was the part just for resilience, as I was under the impression that the edge case of:
  - `time` is exactly at a Period's end
  - next Period is later (there's a discontinuity currently)

Was not properly considered here, but I never reproduced an issue with it.

---

Only risk that I can think of is that we now compute `nextPeriod` before the `periodStreamCleared` callback and `currentCanceller.cancel()` which may perform unknown side-effect.

We could also just re-compute `nextPeriod` there - at the expense of code readability - as you wish.

Reply:
I finally chose to just re-compute it.